test/cql-pytest/util.py

```diff
@@ -248,3 +250,20 @@ def local_process_id(cql):
 # The number of arguments is assumed to be even.
 def user_type(*args):
     return collections.namedtuple('user_type', args[::2])(*args[1::2])
+
+# Tries to inject an error via Scylla REST API. It only works on Scylla,
+# and only in specific build modes (dev, debug, sanitize), so this function
+# will trigger a test to be skipped if it cannot be executed.
+@contextmanager
+def scylla_inject_error(rest_api, err, one_shot=False):
+    url = f"http://{rest_api.host}:{rest_api.port}"
+    response = requests.post(f'{url}/v2/error_injection/injection/{err}?one_shot={one_shot}')
+    response = requests.get(f'{url}/v2/error_injection/injection')
+    print("Enabled error injections:", response.content.decode('utf-8'))
+    if response.content.decode('utf-8') == "[]":
+        pytest.skip("Error injection not enabled in Scylla - try compiling in dev/debug/sanitize mode")
+    try:
+        yield
+    finally:
+        print("Disabling error injection", err)
+        response = requests.delete(f'{url}/v2/error_injection/injection/{err}')
```

Comment:
This was copied from alternator/utils.py right?
If so, a preliminary patch moving it here would be appreciated.
ok with you @nyh?

Reply:
We would like to avoid proliferation of helper utility variants in each test utils "library"....

Reply:
Yes, it was.
Should I delete it from test/alternator/util.py then? That would cause a problem with name shadowing I think. Am I supposed to rename one of util.py files? Or create a new one in test/cql-pytest?

Reply:
It would be best if the alternator test can import it from cql-tests/util.py
but I'm not sure about the exact mechanics.
IIRC we play with the python path in order to import util.py from another directory, so if both alternator and cql-tests have util.py in them that could be a problem.
For the rest_api test suite, we use rest_util.py so maybe we can simply rename alternator/util.py to alternator_util.py so the tests can import both this file and the cql-tests/util.py that can hold the generic library functions.

Does that make sense @nyh ?